### DISCUSSION TITLE
Feature/Env flag for pages

### DISCUSSION THREAD

**Discussion by ccain1-godaddy** [vercel/next.js#29615] [Help]
Is there a way to have either a feature flag or an env config for pages? IE if I want to only have a specific page like `/foo` exist if a flag is true or if in a specific env like dev?

Use case is developing a new page we want to have live in DEV but not released to PROD, for example.

**Comment by zeekrey** [comment]
Maybe you could use the `redirect` prop with `getStaticProps` or `getServerSideProps`:

```javascript
export async function getStaticProps(context) {
  if (procss.env.featureFlag) {
    return {
      redirect: {
        destination: '/',
        permanent: false,
      },
    }
  }

  return {
    props: { data }, // will be passed to the page component as props
  }
}
```

**Reply by joaogarin** [reply]
Does this help the bundle? If the render function of this page still has code/uses packages that other pages also use it could be considered by next as common and so it could happen that still ends up putting javascript inside the common chunks bundle right? Not achieving the real purpose of a feature flag that really removes code from the build folder?

Thinking out loud a little bit but curious if people are using redirect in pages for this purpose. 

And also for specific components how would you tackle this same problem?
